batchflow/models/tf/base.py

```diff
@@ -97,27 +119,51 @@ class TFModel(BaseModel):
         - ``{'loss': external_loss_fn_without_add_loss, 'add_loss': True}``
         - ``{'loss': external_loss_fn_to_collection, 'add_loss': True, 'loss_collection': tf.GraphKeys.LOSSES}``
 
-    decay - a learning rate decay algorithm might be defined in one of three formats:
-        - name
-        - tuple (name, args)
-        - dict {'name': name, **args}
+    optimizer : str, tuple, dict
+        Optimizer, might be defined in multiple formats.
+
+        If str, then short ``name``.
+        If tuple, then ``(name, args)``.
+        If dict, then ``{'name': name, **args}``.
+
+        Name must be one of:
+            - short name (e.g. `'Adam'`, `'Adagrad'`, any optimizer from
+              `tf.train <https://www.tensorflow.org/api_docs/python/tf/train>`_ with or without a word `Optimizer`)
+            - a function name from `tf.train <https://www.tensorflow.org/api_docs/python/tf/train>`_
+              (e.g. 'FtlrOptimizer')
+            - callable
+
+        Examples:
+
+        - ``{'optimizer': 'Adam'}``
+        - ``{'optimizer': ('Ftlr', {'learning_rate_power': 0})}``
+        - ``{'optimizer': {'name': 'Adagrad', 'initial_accumulator_value': 0.01}``
+        - ``{'optimizer': functools.partial(tf.train.MomentumOptimizer, momentum=0.95)}``
+        - ``{'optimizer': some_optimizer_fn}``
+
+    decay : str, tuple, dict
+        Learning rate decay algorithm, might be defined in multiple formats.
 
-        where name might be one of:
+        If str, then short ``name``.
+        If tuple, then ``(name, args)``.
+        If dict, then ``{'name': name, **args}``.
 
-        - short name ('exp', 'invtime', 'naturalexp', 'const', 'poly')
-        - a function name from `tf.train <https://www.tensorflow.org/api_docs/python/tf/train>`_
-          (e.g. 'exponential_decay')
-        - a callable
+        Name must be one of:
+            - short name (e.g. `'exp'`, `'invtime'`, `'naturalexp'`, `'const'`, `'poly'`)
+            - a function name from `tf.train <https://www.tensorflow.org/api_docs/python/tf/train>`_
+              (e.g. 'exponential_decay')
+            - callable
 
         Examples:
 
         - ``{'decay': 'exp'}``
         - ``{'decay': ('polynomial_decay', {'decay_steps': 10000})}``
         - ``{'decay': {'name': tf.train.inverse_time_decay, 'decay_rate': .5}``
 
-    scope - subset of variables to optimize during training. Can be either string or sequence of strings.
-        Value `''` is reserved for optimizing all trainable variables.
-        Putting `-` sign before name stands for complement: optimize everything but the passed scope.
+    scope : str or sequence of str
+        Subset of variables to optimize during training.
+        Value ``''`` is reserved for optimizing all trainable variables.
```

Comment:
Do we have clear rules for using single and double back-ticks?

Reply:
We do!
Double backquotes are reserved for code samples (or things that can appear generally in config), while single ones are used for emphasis